daily_read/order_portal.py

```diff
@@ -68,71 +68,50 @@ def get_orders(self, node=None, status=None, orderer=None, recent=True):
-
-                if proj_info.orderer not in order_updates:
-                    order_updates[proj_info.orderer] = {
-                        "pull_date": f"{pull_date}",
-                        "active_projects": 0,
-                        "recents": {},
-                        "projects": {},
-                    }
-
-                latest_date = "0000-00-00"
-                latest_status = "None"
-                for date_value, date_statuses in proj_info.data["proj_dates"].items():
-                    # Activity on 5 recent dates
-                    recents_keys = sorted(order_updates[proj_info.orderer]["recents"].keys())
-                    if len(recents_keys) < 5:
-                        for status in date_statuses:
-                            order_updates[proj_info.orderer]["recents"].setdefault(date_value, []).append(
-                                (proj_info.data["project_name"], status)
-                            )
+            elif order["identifier"] not in self.projects_data.data.keys():
+                log.debug(f"Order portal id: {order['identifier']} not found in data fetched form sources")
+                continue
+
+            proj_info = self.projects_data.data[order["identifier"]]
+            if order["reports"]:
+                prog_reports = [item for item in order["reports"] if item["name"] == "Project Progress"]
+                if prog_reports:
+                    if len(prog_reports) == 1:
+                        proj_info.report_iuid = prog_reports[0]["iuid"]
                     else:
-                        if date_value > recents_keys[0]:
-                            order_updates[proj_info.orderer]["recents"].pop(recents_keys[0])
-                            for status in date_statuses:
-                                order_updates[proj_info.orderer]["recents"].setdefault(date_value, []).append(
-                                    (proj_info.data["project_name"], status)
-                                )
-
-                    # Find status
-                    if date_value > latest_date:
-                        latest_date = date_value
-                        for status in date_statuses:
-                            if dates_prio[status] > dates_prio[latest_status]:
-                                latest_status = status
-
-                order_updates[proj_info.orderer]["projects"].setdefault(latest_status, []).append(proj_info)
-                order_updates[proj_info.orderer]["active_projects"] += 1
+                        raise ValueError(
+                            f"Multiple reports for Project Progress found in the Order Portal for order {order['identifier']}"
+                        )
+
+            if proj_info.orderer not in order_updates:
+                order_updates[proj_info.orderer] = {
+                    "pull_date": f"{pull_date}",
+                    "active_projects": 0,
+                    "recents": {},
+                    "events": [],
```

Comment:
What will the events be used for?

Reply:
I extract the recents from the events. So it's just intermediate storage